Makefile

```diff
@@ -1,20 +1,30 @@
-#!/usr/bin/env bash
+.PHONY: all clean build swagger test lint run fmt docker-build help
+.DEFAULT_GOAL:=help
 
-.PHONY: all clean build swagger test lint run help
-.DEFAULT_GOAL := help
-
-VERSION := $(shell git describe --always --tags)
-BUILD_TIME=$(shell date '+%Y%m%d-%H%M%S')
+VERSION?=$(shell git describe --always --tags)
+BUILD_TIME?=$(shell date '+%Y%m%d-%H%M%S')
 DOCKER_IMAGE_NAME=spx01/blocky
-BINARY_NAME=blocky
-BIN_OUT_DIR=bin
+
+BINARY_NAME:=blocky
+BIN_OUT_DIR?=bin
+
+GO_BUILD_FLAGS?=-v
+
+GO_BUILD_LD_FLAGS:=\
+	-w \
+	-s \
+	-X github.com/0xERR0R/blocky/util.Version=${VERSION} \
+	-X github.com/0xERR0R/blocky/util.BuildTime=${BUILD_TIME} \
+	-X github.com/0xERR0R/blocky/util.Architecture=${GOARCH}${GOARM}
```

Comment:
`GOARCH` and `GOARM` aren't defined anywhere besides the `Dockerfile`, so this won't work for plain `make`.  
Can you add them below the `BUILD_TIME` definition?

Reply:
Your assumption is wrong as GOARCH & GOARM is set to your system architecture by default.
Its only overwritten in the Dckerfile for crosscompilation purpose.

Reply:
I might have missed something, but to the best of my knowledge, `make` inherits the shell's env, and on my system, those vars are not set. So this would end up with an empty `util.Architecture`.  
Maybe you have a different setup and they get set somehow? But I don't think that's common so not reliable.